report/report.py

```diff
@@ -48,12 +55,133 @@ def report(
         with Session(self.engine) as session:
             session.add(
                 make_iteration(
-                    run_id=self.run_id,
-                    benchmark=self.benchmark,
+                    run_id=run_id,
+                    benchmark=benchmark,
                     iteration_no=iteration_no,
-                    run_params=self.run_params,
+                    run_params=run_params,
                     name2time=name2time,
                     params=params,
                 )
             )
             session.commit()
+
+    def report_arbitrary(
+        self,
+        benchmark: str,
+        backend: str,
+        name2time: Dict[str, float],
+        run_id: int | None = None,
+        iteration_no: int = 1,
+        params: Dict[str, str] | None = None,
```

Comment:
Let's try to keep the order of the parameters similar to what is in `report` func.

```suggestion
        benchmark: str,
        run_id: int | None = None,
        backend: str,
        iteration_no: int = 1,
        name2time: Dict[str, float],
        params: Dict[str, str] | None = None,
```

Reply:
I'll change order of arguments and also add `*` to force keyword-only calls. Funciton with so many arguments should not be called without keywords.

Current suggestion wouldn't work without `*` either way, because arguments with default values precede necessary arguments.